kong/observability/logs.lua

```diff
@@ -0,0 +1,169 @@
+local _M = {
+  maybe_push = function() end,
+  get_request_logs = function() return {} end,
+  get_worker_logs = function() return {} end,
+}
+
+if ngx.config.subsystem ~= "http" then
+  return _M
+end
+
+
+local request_id_get = require "kong.observability.tracing.request_id".get
+local time_ns = require "kong.tools.time".time_ns
+local cycle_aware_deep_copy = require "kong.tools.utils".cycle_aware_deep_copy
+
+local get_log_level = require "resty.kong.log".get_log_level
+local constants_log_levels = require "kong.constants".LOG_LEVELS
+
+local table_new = require "table.new"
+local table_clear = require "table.clear"
+local string_buffer = require "string.buffer"
+
+local ngx = ngx
+local kong = kong
+local table = table
+local tostring = tostring
+
+local MAX_WORKER_LOGS = 1000
+local MAX_REQUEST_LOGS = 1000
+local NGX_CTX_REQUEST_LOGS_KEY = "o11y_logs_request_scoped"
+
+local worker_logs = table_new(100, 0)
+local logline_buf = string_buffer.new()
+
+
+local function configured_log_level()
+  local ok, level = pcall(get_log_level)
+  if not ok then
+    level = constants_log_levels[kong.configuration.log_level]
+  end
+
+  return level
+end
+
+
+-- needed because table.concat doesn't like booleans
+local function concat_tostring(tab)
+  if #tab == 0 then
+    return ""
+  end
+
+  for i = 1, #tab do
+    logline_buf:put(tostring(tab[i]))
```

Comment:
```
buf = buf:put([str|num|obj] [,…])
Appends a string str,
a number num or any object obj with a __tostring metamethod
to the buffer. Multiple arguments are appended in the given order.

Appending a buffer to a buffer is possible
and short-circuited internally. But it still involves a copy.
Better combine the buffer writes to use a single buffer.

https://luajit.org/ext_buffer.html
```

I think the `tostring` is not required here.

Reply:
I thought I tried and it didn't like booleans, I can try again.

Reply:
Looks like it doesn't like booleans: https://github.com/openresty/luajit2/blob/d4c5a405c3aedd75e1f491437aac6af274bcd045/src/lib_buffer.c#L123-L144